src/core/model/reducer.ts

```diff
@@ -57,61 +68,47 @@ export const combinedReducer = (state: EditorState, action: CombinedAction) => {
         schema: buildSchemaTree(action.schema),
         uiSchema: buildLinkedUiSchemaTree(action.uiSchema),
       };
-    case ADD_SCHEMA_ELEMENT_TO_LAYOUT:
-      const newUiSchema = cloneTree(action.layout as LinkedUISchemaElement);
-      const newSchema = cloneTree(action.schemaElement);
-      if (isPathError(newUiSchema)) {
-        console.error(
-          'An error occured when cloning the ui schema',
-          newUiSchema
-        );
-        // Do nothing
-        return state;
-      }
-      if (isPathError(newSchema)) {
-        console.error('An error occured when cloning the schema', newSchema);
-        // Do nothing
-        return state;
-      }
-      const newControl: LinkedUISchemaElement = createControl(
-        `#${getPath(action.schemaElement)}`
+    case ADD_SCOPED_ELEMENT_TO_LAYOUT:
+      return withCloneTree(
```

Comment:
I would prefer to use the `withCloneTree` cascaded in each other, this way the withCloneTree would be simpler, maybe provide a helper to cascade them?